core/appeal/service.go

```diff
@@ -1121,6 +1134,110 @@ func getPolicy(a *domain.Appeal, p *domain.Provider, policiesMap map[string]map[
 	return policy, nil
 }
 
+func (s *Service) populateAppealMetadata(ctx context.Context, a *domain.Appeal, p *domain.Policy) error {
+	if p.AppealMetadataSources == nil {
+		return nil
+	}
+
+	appealMetadata := map[string]interface{}{}
+	for key, metadata := range p.AppealMetadataSources {
+		if metadata.Type == "http" {
+			if metadata.Config.URL == "" {
+				return fmt.Errorf("URL cannot be empty for http type")
+			}
+
+			metadataCl, err := s.getMetadataClient(metadata.Config)
+			if err != nil {
+				return fmt.Errorf("key: %s, %w", key, err)
+			}
+
+			res, err := metadataCl.GetResourceMetadata()
+			if err != nil && !metadata.Config.AllowFailed {
+				return fmt.Errorf("error fetching resource: %w", err)
+			}
+
+			responseMap := res.(map[string]interface{})
+			params := map[string]interface{}{
+				"response": responseMap,
+				"appeal":   a,
+			}
+			value, err := s.getMetadataValue(metadata.Type, key, metadata.Value, params)
+			if err != nil {
+				return fmt.Errorf("error parsing value: %w", err)
+			}
+			appealMetadata[key] = value
+
+		} else if metadata.Type == "static" {
+			params := map[string]interface{}{
+				"appeal": a,
+			}
+			value, err := s.getMetadataValue(metadata.Type, key, metadata.Value, params)
+			if err != nil {
+				return fmt.Errorf("error parsing value: %w", err)
+			}
+			appealMetadata[key] = value
+		}
+	}
+
+	a.Details["__policy_metadata"] = appealMetadata
+
+	return nil
+}
+
+func (s *Service) getMetadataClient(metadataCfg *domain.AppealMetadataConfig) (*identities.HTTPClient, error) {
+	metadataConfig, err := s.metadata.ParseMetadataConfig(metadataCfg)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing metadata config: %w", err)
+	}
+
+	clientConfig, ok := metadataConfig.(*identities.HTTPClientConfig)
+	if ok {
+		return nil, fmt.Errorf("invalid metadata config: %w", err)
+	}
+
+	metadataCl, err := identities.NewHTTPClient(clientConfig)
```

Comment:
ideally, we shouldn't use `identities` because it's dedicated to the creator details and we are planning to replace that with this feature itself. Additionally, metadata will not always be about user details (identity). 
it would be better if we init the `http.Client` directly here or create new package `pkg/http` would be reasonable given we have our own HTTP config abstraction (the `HTTPClientConfig` struct used also should be the one from that new package)
with this approach we can remove dependency from `identities` and also remove the need to have MetadataManager concept altogether